editor/plugins/shader_editor_plugin.cpp

```diff
@@ -413,11 +413,29 @@ void ShaderEditorPlugin::_menu_item_pressed(int p_index) {
 			if (editor && editor->get_trim_trailing_whitespace_on_save()) {
 				editor->trim_trailing_whitespace();
 			}
+
+			Ref<Resource> resource;
 			if (edited_shaders[index].shader.is_valid()) {
-				EditorNode::get_singleton()->save_resource(edited_shaders[index].shader);
+				resource = edited_shaders[index].shader;
 			} else {
-				EditorNode::get_singleton()->save_resource(edited_shaders[index].shader_inc);
+				resource = edited_shaders[index].shader_inc;
 			}
+
+			EditorNode::get_singleton()->save_resource(resource);
+
+			if (resource->get_path().is_resource_file()) {
+				// If resource file exists, this means that the shader will not trigger a save-as popup-up.
```

Comment:
Note that Save As popup is caused by Shader Editor not handling built-in scripts properly. #84630 will resolve this.